gpiozero/internal_devices.py

```diff
@@ -514,51 +544,97 @@ class TimeOfDay(PolledInternalDevice):
         The time after which the device will be considered inactive.
 
     :param bool utc:
-        If :data:`True` (the default), a naive UTC time will be used for the
-        comparison rather than a local time-zone reading.
+        If `None` (the default), UTC time will be used for the comparison. 
+        If `False` the local clock time will be use, ignoring the timezone. 
+        (If `True` a naive UTC time will be used - this is not recommended, 
+        see the note above)
 
     :type event_delay: float
     :param event_delay:
-        The number of seconds between file reads (defaults to 10 seconds).
+        The number of seconds between file reads (defaults to 5 seconds).
 
     :type pin_factory: Factory or None
     :param pin_factory:
         See :doc:`api_pins` for more information (this is an advanced feature
         which most users can ignore).
     """
-    def __init__(self, start_time, end_time, *, utc=True, event_delay=5.0,
+
+    def __init__(self, start_time, end_time, *, utc=None, event_delay=5.0,
```

Comment:
What backwards-compatibility implications does changing the default-value for `utc` have?

Reply:
Sematically, the behaviour is still the same: default behaviour is for the times to be treated as UTC. Technically, those will now be timezone-aware UTC times rather than naive ones.
TimeOfDay will still act exactly as before if the default is used, `True` is passed or `False` is passed.

The only risk I see is if someone has made some kind of use of the fact that previously the times were naive *and* had relied on implicity specifying `utc` rather than explicitly stating `utc=True`. I can't thin of any cases where someone would be doing something with the times which needed them to be naive, other than working in "local time".

Would it be worthwhile rephrasing `self._aware = utc is None` to something which takes a little more space and makes the behaviour more apparent. At the moment most of the relevant logic is later in `_validate_time()`

Reply:
I think I'd need to do more reading up to understand the difference between "timezone-aware UTC times" and "naive UTC times" to understand what you're saying here; which I'm afraid I don't have time to do right now :wink:

Reply:
@bennuttall , @waveform80 - the more I think about this; the more I think it would make sense to take the deprecation a step further and introduce a new boolean keyword argument `local` in addition to the changes to `utc`
I would move `utc` to the end of the signature and documentation, placing `local` more prominently.

This would make the normal usage:
1. use UTC times (in a timezone-aware fashion) = default
2. specify `tzinfo` in the construction of the start & end times to use a specific timezone
3. specify `local=True` to use local time

While still maintaining backwards compatibility, by keeping `utc` semantically identical to now.

@lurch  - I'd also be interested in your views on this if you have any; your comments have been really useful in cleaning up rough edges.